### PR TITLE
pour_bottle?: Pour local bottles without sha256

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Pouring a local bottle for a formula without a bottle sha256
in the formula for that OS would unexpectedly install from
source instead.

### CODE REVIEW

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def pour_bottle?(install_bottle_options = { warn: false })
     return false if @pour_failed
 
     bottle = formula.bottle
-    return false unless bottle
+    return false unless bottle || formula.local_bottle_path
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_installer.rb]: `return true  if formula.local_bottle_path` is on line 94 already so it feels weird to duplicate it here. Can this logic be moved around instead so it's only checked once?

Comment (sjackman) [reply Library/Homebrew/formula_installer.rb]: I've removed `return true if formula.local_bottle_path`. Pouring a local bottle should still check that that bottle is compatible. `--force-bottle` may be used to pour a local bottle that is incompatible, as it is for remote bottles.

Comment (MikeMcQuaid) [reply Library/Homebrew/formula_installer.rb]: Ok, cool. Can you make this an `if` instead; `unless` with multiple conditions are hard to parse. After that should be 👍 to 🚢

Comment (sjackman) [reply Library/Homebrew/formula_installer.rb]: Done. Thanks for the feedback, Mike.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Comment by sjackman** [comment]
My pleasure. Thanks for the feedback, and thanks for merging, Mike.
